python/ray/autoscaler/ray-schema.json

```diff
@@ -156,6 +156,36 @@
                             "description": "Oauth token or JSON string constituting service account credentials"
                         }
                     }
+                },
+                "cloudwatch": {
+                    "agent": {
+                        "config": {
+                            "type": ["string", "null"],
+                            "description": "Path to CloudWatch Agent config file. See https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/CloudWatch-Agent-Configuration-File-Details.html for additional details."
+                        },
+                        "retryer": {
+                            "max_attempts": {
+                                "type": ["integer", "null"],
+                                "description": "Max allowed CloudWatch agent installation attempts on any host."
+                            },
+                            "delay_seconds": {
+                                "type": ["integer", "null"],
+                                "description": "Seconds to wait between each CloudWatch agent installation attempt."
+                            }
+                        }
+                    },
+                    "dashboard": {
+                        "config": {
+                            "type": ["string", "null"],
+                            "description": "Path to CloudWatch Dashboard config file. See https://docs.aws.amazon.com/AmazonCloudWatch/latest/APIReference/CloudWatch-Dashboard-Body-Structure.html for additional details."
+                        }
+                    },
+                    "alarm": {
+                        "config": {
+                            "type": ["string", "null"],
+                            "description": "Path to CloudWatch Alarm config file. See https://docs.aws.amazon.com/AmazonCloudWatch/latest/APIReference/API_PutMetricAlarm.html for additional details."
```

Comment:
IMO, it's not obvious that this is expected to be an array of PutMetricAlarm requests.

Reply:
Agreed. Perhaps we could make this a more clear by changing the config section heading to from "alarm" to "alarms" and explicitly noting this in the description (e.g. "Path to CloudWatch config file containing an array of JSON PutMetricAlarm requests. See https://docs.aws.amazon.com/AmazonCloudWatch/latest/APIReference/API_PutMetricAlarm.html for additional details.").